### DISCUSSION TITLE
Could it be that the the misleading certificate count issues are influenced by covpass app feature

### DISCUSSION THREAD

**Discussion by grimch** [Digitaler-Impfnachweis/covpass-android#179] [Q&A] [answered]
I saw the discussion in the news today about some 40 million certificates more issued than vaccinations given:
https://www.spiegel.de/netzwelt/web/covid-impfung-42-millionen-ueberschuessige-impfzertifikate-ausgestellt-a-fd97dbca-8e8c-4bc9-9210-889416e97fb9

It reminded me of some functionality in the CovPass app to generate a print out "EU-Ausdruck erstellen". I have used the same at one time to have pdf-file of my certificate, which I have stored separately.

I don't know how you are generating that pdf but if you to a round trip to the RKI servers, they **might increase a particular counter** in the same way as it is happening, when such a certificate document is first generated e.g. by a pharmacy after the vaccination.

If the latter is the case, then it would explain the massive surplus numbers at least partially. 
Therefore my question: When you are creating these print out pdfs, are you interacting with the RKI server for that?

**Comment by Ein-Tim** [comment]
I'm sure @timokoenig knows the answer here.

**Comment by timokoenig** [comment, marked as answer]
@grimch CovPass does not share any data. The app generates the PDF on the device so it would not affect the count of the certificates.

There are quite a few reasons why there are more certificates than vaccinations:
* The certificate has a technical expiration
* The data like the dose number on the certificate is wrong
* The users name is wrong or has changed (marriage)
* The user lost the certificates

All the mentioned cases and possibly some other would result in one or more reissued certificates per user

**Comment by vaubaehn** [comment]
@grimch and @timokoenig 

The points mentioned by Timo are totally plausible. But it still leaves the question open (for me), why there is a "sudden" increasing gap since the middle of December 2021, as stated in the Spiegel article (shared from @grimch ) and also stated here: https://www.n-tv.de/panorama/Viel-mehr-Zertifikate-als-Impfungen-article23086569.html .

Purely speculating, but I could imagine that middle December was the point in time when the Booster-Campaign started in doctors' offices. In one CWA-issue about failed grouping of DCCs in the wallet apps @oliver-steinbrecher stated, that the health insurance card is used for taking the data to create the DCC in an automated process without further user interaction. But this data structure of the insurance card differs from the data that is requested in pharmacies or vaccination centers (https://github.com/corona-warn-app/cwa-wishlist/issues/705#issuecomment-1017973022). @thinkberg implied something similarly (https://github.com/corona-warn-app/cwa-quick-test-frontend/issues/185#issuecomment-1021501696) . So there is a certain likelihood that boosters can't be grouped with vaccination DCCs due to different data in the standardized name fields. This might lead to many users getting their former vacc DCCs (or the booster DCC) re-issued. In this case, only a survey among pharmacies could reveal some evidence what actually happened.

Is that a plausible hypothesis for what happened since mid December?

**Reply by timokoenig** [reply]
I think one of the main reasons why it increased recently is the booster campaign and the 2G/2G+ rules. Every time something was wrong (grouping, dosage, ...) the communication was "go the the pharmacy and get a new certificate issued".

Booster certificates are still one of the biggest issues. Just today I noticed a person going to multiple pharmacies getting the same certificate issued twice because it was not usable. The person has recovered and just got a booster but the certificate stated 2/2 which is without further knowledge a full immunization and not a booster. Therefore the certificate is only valid after 14 days which is clearly wrong. The numbers will increase further if these small problems will not get fixed asap.

**Reply by vaubaehn** [reply]
Thanks for your reply. The reasons you mentioned possibly contribute most to "the issue". As far what I understood from many discussions on GH, missing information in context of 2G/2G+ rules (i.e., how are admissions rules and immunization states are defined between federal states) is also a problem for app development ("booster detection"). Tim and me are planning to reach out to RKI to address some of these issues. We should pick up that topic here ("high pharmacy usage for re-issuing") also - agreed, Tim?

**Reply by Ein-Tim** [reply]
Agreed. (-:

**Reply by oliver-steinbrecher** [reply]
DCC Naming and grouping is independent from boosters. It is still an issue but definitely hard to fix at the moment. 
Starting in Februar there will be new EU requirements on how to encode boosters. This  will improve the situation a bit

**Reply by ndegendogo** [reply]
> Starting in Februar there will be new EU requirements on how to encode boosters. 

@oliver-steinbrecher will this then require a re-issuance of certificates with "legacy-encoded" booster?

**Reply by timokoenig** [reply]
@ndegendogo I would guess this could be true for all booster certificates that are not clearly identifiable.

One thing I want to mention is that regardless of the new booster encoding, the EU and RKI/BMG need to be aligned for future regulation changes. In the beginning of the pandemic we had the first issue with the "full immunization status" where we had problems to identify people and their 14 day period. Then regulation changed and we had problems identifying recovered full immunizations. Then regulation changed again and we had problems identifying booster certificates. Now we have "a solution" for the booster problem but is this the end? I don't think so. Who knows what will come in the future but maybe the next thing will be to identify booster certificates of a booster and if the RKI/BMG is not aligned with the current state of the EU DCC then we will have the next issue that will haunt us for month. All I am saying is that regulations can only change when all possible cases are achievable with the current state of the EU DCC.

**Reply by oliver-steinbrecher** [reply]
> > Starting in Februar there will be new EU requirements on how to encode boosters.
> 
> @oliver-steinbrecher will this then require a re-issuance of certificates with "legacy-encoded" booster?

We’re working on a solution.
